src/components/react-json-schemaForm/GeometryField/GeometryField.js

```diff
@@ -1,48 +1,219 @@
-    <div className="form-group field">
-      {children}
-    </div>
-  </fieldset>
-);
+const getLayerPaintId = (layerPaints, name) => {
+  const { id } = layerPaints.find(({ 'source-layer': source }) => source === name) || {};
+  return id;
+};
 
-const GeometryField = props => {
-  const { formData: { coordinates, type }, t } = props;
-  if (type === 'Point') {
-    return (
-      <Fieldset t={t}>
-        <PointField {...props} />
-      </Fieldset>
-    );
+const resetStyle = (map, layerPaints, name) => {
+  if (!Object.keys(map).length) {
+    return;
+  }
+  const layerPaintId = getLayerPaintId(layerPaints, name);
+  map.setFilter(layerPaintId, null);
+};
+
+/**
+ * Get geometries from
+ * -settings informations if this is a new one
+ * -feature's stored if this is an update
+ *
+ * @param {Object} {
+ *   feature,
+ *   formData: { coordinates, type },
+ *   params: { id, layer },
+ *   name,
+ *   settings,
+ * }
+ * @returns an object wrapping
+ * the name, identifier, geometry and endpoint (to save the feature) to manipulate
+ */
+const getGeometries = ({
+  feature,
+  formData: { coordinates, type },
+  params: { id, layer },
+  name,
+  settings,
+}) => {
+  if (id === ACTION_CREATE) {
+    const { layer: { name: layerName, ...rest }, featureEndpoint } = getView(settings, layer);
+    return {
+      ...rest,
+      identifier: null,
+      geom: { coordinates, type },
+      layerName,
+      url: featureEndpoint,
+    };
   }
-  const action = !coordinates.length ? 'create' : 'edit';
+  const { geometries: { [name]: featureName } } = feature[id];
+  return { layerName: name, ...featureName };
+};
+
+
+const GeometryField = props => {
+  const {
+    addControl,
+    feature,
+    formData,
+    layerPaints,
+    map,
+    match: { params },
+    removeControl,
+    name,
+    settings,
+    t,
+    ...rest
+  } = props;
+
+  const geometries = useMemo(() => (
+    getGeometries(({
+      feature,
+      formData,
+      params,
+      name,
+      settings,
+    }))
+  ), [feature, formData, params, name, settings]);
+
+  const [geomValues, setGeomValues] = useState(geometries);
+
+  useEffect(() => {
+    setGeomValues(geometries);
+  // Disabling `geometries` because it's only necessary updating value on mount
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [params]);
+
+  const updateGeometryFromMap = useCallback(({
+    features: [{ geometry, id }],
+    target,
+    type,
+  }) => {
+    if (type === 'draw.delete') {
+      setGeomValues(prevGeomValues => ({
+        ...prevGeomValues,
+        geom: null,
+      }));
+      return;
+    }
+
+    const { features } = target.draw.getAll();
+
+    if (features.length > 1 && [POINT, LINESTRING, POLYGON].includes(geomValues.geom_type)) {
+      target.draw.delete(
+        features.reduce((list, item) => (
+          item.id !== id
+            ? [...list, item.id]
+            : list
+        ), []),
+      );
```

Comment:
Doing this in two steps improve understandability.
*(also, simpler case first)*

```suggestion
      const otherFeaturesIds = features.reduce((list, item) => (
        item.id === id ? list : [...list, item.id]
      ), []);
        
      target.draw.delete(otherFeaturesIds);
```